File: core/src/galenscovell/oregontrail/ui/screens/AbstractScreen.java
package galenscovell.oregontrail.ui.screens;

import com.badlogic.gdx.*;
import com.badlogic.gdx.graphics.GL20;
import com.badlogic.gdx.scenes.scene2d.Stage;
import galenscovell.oregontrail.OregonTrailMain;

public class AbstractScreen implements Screen {
    protected final OregonTrailMain root;
    protected Stage stage;

    public AbstractScreen(OregonTrailMain root) {
        this.root = root;
    }

    public void create() { }

    @Override
    public void render(float delta) {
        Gdx.gl.glClearColor(0, 0, 0, 1);
        Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
        stage.act(delta);
        stage.draw();
    }

    @Override
    public void resize(int width, int height) {
        if (stage != null) {
            stage.getViewport().update(width, height, true);
        }
    }

    @Override
    public void show() {
        create();
        Gdx.input.setInputProcessor(stage);
    }

    @Override
    public void hide() {
        Gdx.input.setInputProcessor(null);
    }

    @Override
    public void pause() { }

    @Override
    public void resume() { }

    @Override
    public void dispose() {
        if (stage != null) {
            stage.dispose();
        }
    }
}

File: core/src/galenscovell/oregontrail/ui/screens/AbstractScreen.scala
package galenscovell.oregontrail.ui.screens

import com.badlogic.gdx._
import com.badlogic.gdx.graphics.GL20
import com.badlogic.gdx.scenes.scene2d.Stage
import galenscovell.oregontrail.OregonTrailMain


class AbstractScreen(val gameRoot: OregonTrailMain) extends Screen {
  protected final val root: OregonTrailMain = gameRoot
  protected var stage: Stage = null


  protected def create(): Unit = {

  }

  override def render(delta: Float): Unit = {
    Gdx.gl.glClearColor(0, 0, 0, 1)
    Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT)
    stage.act(delta)
    stage.draw()
  }

  override def resize(width: Int, height: Int): Unit = {
    if (stage != null) {
      stage.getViewport.update(width, height, true)
    }
  }

  override def show(): Unit = {
    create()
    Gdx.input.setInputProcessor(stage)
  }

  override def hide(): Unit = {
    Gdx.input.setInputProcessor(null)
  }

  override def pause(): Unit =  {

  }

  override def resume(): Unit =  {

  }

  override def dispose(): Unit = {
    if (stage != null) {
      stage.dispose()
    }
  }
}
